gm4_soul_fragments/data/gm4_soul_fragments/functions/player/submain.mcfunction

```diff
@@ -0,0 +1,9 @@
+# @s = player that has lost soul fragments
+# at @s
+# from main
+
+# check for missing soul fragments and signs of life for regeneration managing
+execute as @a[scores={gm4_sf_fragments=1..}] unless score @s gm4_sf_rspwn matches 1 at @s run function gm4_soul_fragments:player/manage_regen
```

Comment:
shouldn't this be an `@s`, not an `@a`? And at that point you should just use `if score` again
Also, you don't need this `at @s` here, because you do `at @s` in `manage_regen`